src/core/lv2/Lv2Proc.cpp

```diff
@@ -419,11 +436,47 @@ bool Lv2Proc::hasNoteInput() const
 
 
 
+void Lv2Proc::initMOptions()
+{
+	const Lv2UridCache& cache = Engine::getLv2Manager()->uridCache();
+
+	uint32_t atom_Float = cache[Lv2UridCache::Id::atom_Float];
+	uint32_t atom_Int = cache[Lv2UridCache::Id::atom_Int];
+
+	/*
+		sampleRate:
+		LMMS can in theory inform plugins of a new sample rate.
+		However, Lv2 plugins seem to not allow sample rate changes
+		(not even through LV2_Options_Interface) - it's assumed to be
+		fixed after being passed via LV2_Descriptor::instantiate.
+		So, if the sampleRate would change, the plugin will need to
+		re-initialize, and this code section will be
+		executed again, creating a new option vector.
+	*/
+	float sampleRate = Engine::mixer()->processingSampleRate();
+	int32_t blockLength = Engine::mixer()->framesPerPeriod();
+	int32_t sequenceSize = defaultEvbufSize();
+
+	m_options.initOption(cache[Lv2UridCache::Id::param_sampleRate],
+		atom_Float, &sampleRate);
+	m_options.initOption(cache[Lv2UridCache::Id::bufsz_maxBlockLength],
+		atom_Int, &blockLength);
+	m_options.initOption(cache[Lv2UridCache::Id::bufsz_minBlockLength],
+		atom_Int, &blockLength);
+	m_options.initOption(cache[Lv2UridCache::Id::bufsz_nominalBlockLength],
+		atom_Int, &blockLength);
+	m_options.initOption(cache[Lv2UridCache::Id::bufsz_sequenceSize],
+		atom_Int, &sequenceSize);
```

Comment:
If you pass values by their addresses and never copy them, you will get undefined behavior after this function returns.

Reply:
This could even be the reason for the issues that @zonkmachine has and I previously had. Thanks for the hint.

Reply:
That was my guess, too.

Reply:
Should be solved with 78d8fac.